src/ethereum/mod.rs

```diff
@@ -37,12 +43,41 @@ pub struct Ethereum {
     polling_interval: Duration,
     /// A handle to the event loop that runs mosaic.
     event_loop: Box<tokio_core::reactor::Handle>,
+    /// List of block observers. These are notified when any new block is generated.
+    observers: Vec<Reactor>,
+}
+
+/// This enum represents all reactors which will react to block generation.
+#[derive(Debug, Clone)]
+pub enum Reactor {
+    BlockReporter { block_store_address: Address, validator_address: Address },
 }
 
 trait IntoBlock {
     fn into_block(&self) -> Result<Block, Error>;
 }
 
+/// Anything that wants to react on block generation should implement this.
+trait Observe {
+    fn observe(&self, value: &Block, block_chain: &Ethereum);
+}
+
+impl Observe for Reactor {
+    fn observe(&self, block: &Block, block_chain: &Ethereum) {
```

Comment:
Please add function documentation comments.
Also, this could possibly go into the observer module. Not sure, though.